lib/linter/linter.js

```diff
@@ -1326,19 +1326,39 @@ class Linter {
      */
     constructor({ cwd, configType = "flat", flags = [] } = {}) {
 
+        const processedFlags = [];
+
         flags.forEach(flag => {
             if (inactiveFlags.has(flag)) {
-                throw new Error(`The flag '${flag}' is inactive: ${inactiveFlags.get(flag)}`);
+                const inactiveFlagData = inactiveFlags.get(flag);
+                const inactivityReason = getInactivityReasonMessage(inactiveFlagData);
+
+                if (typeof inactiveFlagData.replacedBy === "undefined") {
+                    throw new Error(`The flag '${flag}' is inactive: ${inactivityReason}`);
+                }
+
+                if (typeof inactiveFlagData.replacedBy === "string") {
```

Comment:
Just to add a bit of clarity.

```suggestion
                // if there's a replacement, enable it instead of original
                if (typeof inactiveFlagData.replacedBy === "string") {
```